recipes/log4cplus/all/conanfile.py

```diff
@@ -37,7 +37,7 @@ class Log4cplusConan(ConanFile):
 
     def requirements(self):
         if self.options.with_iconv:
-            self.requires.add('libiconv/1.15')
+            self.requires.add('libiconv/1.16')
```

Comment:
```suggestion
            self.requires('libiconv/1.16')
```

Reply:
is it a big deal? both works from my experience. if `self.requires.add` has some disadvantages, we might need to add a hook check
/cc @uilianries

Reply:
Yes, it is. `self.requires.add` is not documented and can change at any moment. We don't want people copy/pasting from a wrong recipe in CCI. Here it happened with `build_requires`: https://github.com/conan-io/conan/issues/6667

Reply:
that's documented (at least I can find it in docs, e.g. [here](https://conan-docs.readthedocs.io/en/feature-cmake-toolchain-file/howtos/create_installer_packages.html))

Reply:
That `https://conan-docs.readthedocs.io/` link belongs to a 0.XX version of the documentation...

Reply:
self.requires.add is outdated